src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,186 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { Todo } from '../../types/todo';
+import { TodoList } from '../TodoList';
+import { Header } from '../Header';
+import { SortType } from '../../types/sort';
+import { Footer } from '../Footer';
+
+function useLocaleStorage<T>(
+  key: string,
+  initialValue: T,
+): [T, (value: T) => void] {
+  const [value, setValue] = useState<T>(() => {
+    try {
+      return JSON.parse(localStorage.getItem(key) || '') || initialValue;
+    } catch (error) {
+      return initialValue;
+    }
+  });
+
+  const save = (saveValue: T) => {
+    setValue(saveValue);
+    localStorage.setItem(key, JSON.stringify(saveValue));
+  };
+
+  return [value, save];
+}
+
+export const TodoApp: React.FC = () => {
+  const [todos, setTodos] = useLocaleStorage<Todo[]>('todos', []);
+  const [title, setTitle] = useState('');
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+
+  const { status } = useParams();
+
+  useEffect(() => {
+    setVisibleTodos(todos.filter(todo => {
+      switch (status) {
+        case SortType.Completed:
+          return todo.completed;
+
+        case SortType.Active:
+          return !todo.completed;
+
+        default:
+          return todo;
+      }
+    }));
+  }, [todos, status]);
+
+  const getActiveTodosCount = () => {
```

Comment:
You can move this out of component and pass todos as an argument

Reply:
and it is not clear for me why do you need to pass this callback (useCallback for such cases)
but you can calculate this count here and pass number to child component

Reply:
> and it is not clear for me why do you need to pass this callback (useCallback for such cases) but you can calculate this count here and pass number to child component

@Bohdan-Maksymiuk I'm sorry, somehow I've missed it